readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/writer/PartitionedWriter.java

```diff
@@ -190,13 +203,18 @@ private void possiblyCloseConsumers(Predicate<ExpiringConsumer> shouldClose) {
             perPartitionDayWriters.forEach((partitionId, dailyWriters) ->
                 dailyWriters.entrySet().removeIf(entry -> {
                     final ExpiringConsumer<MESSAGE_KIND> consumer = entry.getValue();
+                    final LocalDateTime day = entry.getKey();
 
                     if (shouldClose.test(consumer)) {
                         if (tryExpireConsumer(consumer)) {
                             final Counter.Child filesCommitted = PrometheusMetrics.buildCounterChild(
                                 PrometheusMetrics.FILES_COMMITTED, eventName);
 
                             filesCommitted.inc();
+
+                            checkpointer.tryCheckpoint(partitionId, latestMessageTimeForPartitionAndDay.get(
```

Comment:
if checkpointing fails due to an exception, we want to continue and try to checkpoint at the next commit right ?
If so we should protect with try/catch.

Also, I would like to add a prometheus gauge to create an alert on checkpointing not performed for a partition/day

Reply:
Done